engine/crates/engine-v2/src/response/read/view/de.rs

```diff
@@ -0,0 +1,175 @@
+    where
+        V: Visitor<'de>,
+    {
+        visitor.visit_unit()
+    }
+
+    forward_to_deserialize_any! {
+        bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+        bytes byte_buf unit unit_struct newtype_struct seq tuple
+        tuple_struct map struct enum identifier
+    }
+}
+
+impl<'de> IntoDeserializer<'de, InputValueSerdeError> for ResponseObjectsView<'de> {
+    type Deserializer = Self;
+    fn into_deserializer(self) -> Self::Deserializer {
+        self
+    }
+}
+
+impl<'de> serde::Deserializer<'de> for ResponseObjectView<'de> {
+    type Error = InputValueSerdeError;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        MapDeserializer::new(self.selection_set.into_iter().map(|selection| {
+            let key = self.schema[selection.name].as_str();
+            let value = ResponseValueWalker {
+                schema: self.schema,
+                response: self.response,
+                value: self.response_object.find(selection.edge).unwrap_or(&NULL),
+                selection_set: &selection.subselection,
+            };
+            (key, value)
+        }))
+        .deserialize_any(visitor)
+    }
+
+    fn deserialize_option<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        visitor.visit_some(self)
+    }
+
+    fn deserialize_ignored_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        visitor.visit_unit()
+    }
+
+    forward_to_deserialize_any! {
+        bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+        bytes byte_buf unit unit_struct newtype_struct seq tuple
+        tuple_struct map struct enum identifier
+    }
+}
+
+impl<'de> IntoDeserializer<'de, InputValueSerdeError> for ResponseObjectView<'de> {
+    type Deserializer = Self;
+    fn into_deserializer(self) -> Self::Deserializer {
+        self
+    }
+}
+
+impl<'de> serde::Deserializer<'de> for ResponseValueWalker<'de> {
+    type Error = InputValueSerdeError;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        match self.value {
+            ResponseValue::Null => visitor.visit_none(),
+            ResponseValue::Boolean { value, .. } => visitor.visit_bool(*value),
+            ResponseValue::Int { value, .. } => visitor.visit_i32(*value),
+            ResponseValue::BigInt { value, .. } => visitor.visit_i64(*value),
+            ResponseValue::Float { value, .. } => visitor.visit_f64(*value),
+            ResponseValue::String { value, .. } => visitor.visit_borrowed_str(value),
+            ResponseValue::StringId { id, .. } => visitor.visit_borrowed_str(&self.schema[*id]),
+            ResponseValue::Json { value, .. } => value
+                .clone()
+                .deserialize_any(visitor)
```

Comment:
Can you not avoid this clone?

Reply:
I could handle a few special cases like String/Number as JSON is used as fallback for any scalar we don't understand. Eventually, I would like to get rid of it by making `ResponseValue` a superset of it, but well. not there yet. Can't avoid the copy here otherwise

Reply:
Is this not a call to `Deserializer::deserialize_any` then?  Because it seems [like `&Value` implements that](https://docs.rs/serde/1.0.203/x86_64-unknown-linux-gnu/serde/de/trait.Deserializer.html#tymethod.deserialize_any)? Or am I missing something?

Reply:
wow ok you're right, good catch. Didn't work because it's `&Box<Value>` so had to do `as_ref()`